### ISSUE TITLE
Cannot compile with previous command after #4613

### ISSUE THREAD

**Issue by maki49** [deepmodeling/abacus-develop#4636]
### Describe the bug

Since 9350441c7 (#4613),  an error occurs when I compile ABACUS with the previous command:
```
-- Found git: attempting to get commit info...
-- Current commit hash: 9350441c7
-- Last commit date: Wed Jul 10 21:28:59 2024 +0800
-- CUDA components detected, but USE_CUDA is set to OFF. NOT building CUDA version of ABACUS.
-- MKL_ARCH: intel64
-- MKL_LINK: dynamic
CMake Error at /usr/local/lib/cmake/mkl-2023.1.0/MKLConfig.cmake:103 (message):
  Invalid MKL_INTERFACE_FULL `intel_lp64`, options are: intel_ilp64
Call Stack (most recent call first):
  /usr/local/lib/cmake/mkl-2023.1.0/MKLConfig.cmake:168 (mkl_message)
  /usr/local/lib/cmake/mkl-2023.1.0/MKLConfig.cmake:296 (define_param)
  cmake/FindMKL.cmake:7 (find_package)
  CMakeLists.txt:398 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/fortneu49/abacus-develop/build/CMakeFiles/CMakeOutput.log".
```

Command: 
```
cmake -B build -DCMAKE_CXX_COMPILER=icpx -DMPI_CXX_COMPILER=mpiicpc -DELPA_DIR=/home/fortneu49/soft/elpa/2021.05.002 -DCereal_INCLUDE_DIR=/home/fortneu49/soft/cereal/include -DBUILD_TESTING=1 -DENABLE_LIBXC=1 -DENABLE_LIBRI=1 && cmake --build build   -j32
```

@caic99 How should I adjust my command after your fix?

**Comment by caic99** [comment]
@maki49 I'm not sure. How did you install MKL?

**Comment by maki49** [comment]
Seems I'm using MKL with oneAPI, there's a `mkl_link_tool` file in `/opt/intel/oneapi/mkl/2023.1.0/bin/intel64`.
`mkl_link_tool -v` returns: 
```
    Intel(R) oneAPI Math Kernel Library (oneMKL) Link Tool v6.3
    ===========================================================

Unknown mode. Specify -libs, -opts, -env or compilation line

Configuration
=============

OS:                     lnx
Architecture:           intel64
Compiler:               intel_f
Linking:                dynamic
Interface layer:        lp64
Parallel:               yes
OpenMP library:         iomp5
```

**Comment by caic99** [comment]
@maki49 would you attach the MKLConfig.cmake file mentioned above? thanks.

**Comment by Qianruipku** [comment]
I cannot compile, either.
![image](https://github.com/deepmodeling/abacus-develop/assets/76200646/92e015b5-e352-4ab0-818c-45dc1250430f)

**Comment by maki49** [comment]
[CMakeOutput.log](https://github.com/user-attachments/files/16171560/CMakeOutput.log)

**Comment by maki49** [comment]
> @maki49 would you attach the MKLConfig.cmake file mentioned above? thanks.

I find the `MKLConfig.cmake` file in `/opt/intel/oneapi/mkl/2023.1.0/lib/cmake/mkl/`. How to attach it?

**Comment by caic99** [comment]
> I find the `MKLConfig.cmake` file in `/opt/intel/oneapi/mkl/2023.1.0/lib/cmake/mkl/`. How to attach it?

"Attach files" button

**Comment by maki49** [comment]
[MKLConfig.cmake.txt](https://github.com/user-attachments/files/16171599/MKLConfig.cmake.txt)

(I add the suffix `.txt` because `.cmake` file is not supported.)
![image](https://github.com/deepmodeling/abacus-develop/assets/33978601/4ff31b64-22b6-4916-bc15-256cc48bf10b)

**Comment by caic99** [comment]
> [MKLConfig.cmake.txt](https://github.com/user-attachments/files/16171599/MKLConfig.cmake.txt)

@maki49 I'm unable to download that file - it shows "not found". Could you please retry uploading the file?

**Comment by caic99** [comment]
For the users having connection problems, we [recommend](https://abacus.deepmodeling.com/en/latest/quick_start/easy_install.html#container-deployment) using the image at `registry.dp.tech/deepmodeling/abacus-intel`.

**Comment by caic99** [comment]
> /usr/local/lib/cmake/mkl-2023.1.0/MKLConfig.cmake:103

@maki49 Hold on - why this file is both under `/usr/local/lib` and `/opt/intel`? I guess there might be something wrong with your installation.

**Comment by maki49** [comment]
[MKLConfig.cmake.txt](https://github.com/user-attachments/files/16171709/MKLConfig.cmake.txt)


There seems nothing different between the two files
![image](https://github.com/deepmodeling/abacus-develop/assets/33978601/3ba2fa79-ee10-46c3-9478-243167108d5d)

**Comment by caic99** [comment]
@Qianruipku  Would you try add a line
`set(ENABLE_SCALAPACK ON)`
in `CMakeLists.txt` before the line
`set(MKL_INTERFACE lp64)`? Many Thanks.

**Comment by caic99** [comment]
@maki49 Sorry, that's another issue. Would you try using `-DMPI_CXX_COMPILER=mpiicpx`?

**Comment by maki49** [comment]
> @maki49 Sorry, that's another issue. Would you try using `-DMPI_CXX_COMPILER=mpiicpx`?

The error messaged changed to: 
```
-- Could NOT find MPI_CXX (missing: MPI_CXX_LIB_NAMES MPI_CXX_HEADER_DIR MPI_CXX_WORKS) 
CMake Error at /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find MPI (missing: MPI_CXX_FOUND)
Call Stack (most recent call first):
  /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.22/Modules/FindMPI.cmake:1830 (find_package_handle_standard_args)
  CMakeLists.txt:236 (find_package)
```
Seems there're no mpiicpx on my machine. Is it because my version of Intel MPI is too old?
![image](https://github.com/deepmodeling/abacus-develop/assets/33978601/78fb6c1b-492e-4ff3-89a9-34ec080a8bc4)

**Comment by AsTonyshment** [comment]
I'm encountering with this compiling issue as well, my compile command is 
`cmake -B build -DMPI_CXX_COMPILER=mpiicpx -DCMAKE_CXX_COMPILER=icpx -DENABLE_LIBXC=1 -DENABLE_PAW=1 -DENABLE_LIBRI=1 -DENABLE_DEEPKS=1 -DTorch_DIR=~/Softwares/libtorch/share/cmake/Torch/ -Dlibnpy_INCLUDE_DIR=~/Softwares/libnpy/include`, which is usable till yesterday.
Error message occurs:

> -- CUDA components detected, but USE_CUDA is set to OFF. NOT building CUDA version of ABACUS.
-- Upgrade to CMake version 3.25.2 or later for native support of Intel compiler(s) icx, icpx. You are running version 3.22.1.
-- MKL_VERSION: 2024.0.0
-- MKL_ROOT: /opt/intel/oneapi/mkl/2024.0
-- MKL_SYCL_ARCH: intel64
-- MKL_ARCH: intel64
-- MKL_SYCL_LINK: dynamic
-- MKL_LINK: dynamic
-- MKL_SYCL_INTERFACE_FULL: intel_ilp64
-- MKL_INTERFACE_FULL: intel_lp64
-- MKL_SYCL_THREADING: tbb_thread
-- MKL_THREADING: intel_thread
-- Found /opt/intel/oneapi/mkl/2024.0/lib/libmkl_intel_lp64.so
-- Found /opt/intel/oneapi/mkl/2024.0/lib/libmkl_intel_thread.so
-- Found /opt/intel/oneapi/mkl/2024.0/lib/libmkl_core.so
-- Found /opt/intel/oneapi/mkl/2024.0/lib/libmkl_sycl_blas.so
-- Found /opt/intel/oneapi/mkl/2024.0/lib/libmkl_sycl_lapack.so
-- Found /opt/intel/oneapi/mkl/2024.0/lib/libmkl_sycl_dft.so
-- Found /opt/intel/oneapi/mkl/2024.0/lib/libmkl_sycl_sparse.so
-- Found /opt/intel/oneapi/mkl/2024.0/lib/libmkl_sycl_data_fitting.so
-- Found /opt/intel/oneapi/mkl/2024.0/lib/libmkl_sycl_rng.so
-- Found /opt/intel/oneapi/mkl/2024.0/lib/libmkl_sycl_stats.so
-- Found /opt/intel/oneapi/mkl/2024.0/lib/libmkl_sycl_vm.so
-- Found /opt/intel/oneapi/mkl/2024.0/lib/libmkl_intel_ilp64.so
-- Found /opt/intel/oneapi/mkl/2024.0/lib/libmkl_tbb_thread.so
-- Found /opt/intel/oneapi/compiler/2024.0/lib/libiomp5.so
-- Updating submodules
-- Updating submodules
-- Updating submodules
-- Found Libxc: version 5.1.7
-- Configuring done
CMake Error at CMakeLists.txt:145 (add_executable):
  Target "abacus" links to target "MKL::MKL_SCALAPACK" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
CMake Error at CMakeLists.txt:145 (add_executable):
  Target "abacus" links to target "MKL::MKL_SCALAPACK" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
CMake Error at deps/libpaw_interface/test/CMakeLists.txt:74 (add_executable):
  Target "paw_lib" links to target "MKL::MKL_SCALAPACK" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
CMake Error at deps/libpaw_interface/test/CMakeLists.txt:74 (add_executable):
  Target "paw_lib" links to target "IntelMKL::MKL" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
CMake Error at source/module_cell/module_paw/CMakeLists.txt:1 (add_library):
  Target "paw" links to target "MKL::MKL_SCALAPACK" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
gmake: *** [Makefile:1054：cmake_check_build_system] 错误 1

**Comment by AsTonyshment** [comment]
> @maki49 Sorry, that's another issue. Would you try using `-DMPI_CXX_COMPILER=mpiicpx`?

I am also using `mpiicpx`, but I cannot compile either.

**Comment by caic99** [comment]
> Seems there're no mpiicpx on my machine. Is it because my version of Intel MPI is too old?

I guess so. You can try updating your library.

**Comment by caic99** [comment]
@Qianruipku @AsTonyshment Hi there, would you try if #4644 patching works? Thanks.

**Comment by AsTonyshment** [comment]
Sorry, I broke my environment while trying to upgrade my Intel OneAPI Toolkit, and my system even cannot find `icpx` at the momment🥲 I am trying to reinstall the environment now...

**Comment by Cstandardlib** [comment]
> Sorry, I broke my environment while trying to upgrade my Intel OneAPI Toolkit, and my system even cannot find `icpx` at the momment🥲 I am trying to reinstall the environment now...

Maybe you can try `Repair` the installation (comes with oneAPI installer), it will try to fix the environment and solve conflicts between distributions.

**Comment by AsTonyshment** [comment]
I have successfully upgraded my MKL version to 2024.2.0 and pulled the latest repository of ABACUS. I am now able to compile ABACUS successfully, but only when the LIB_PAW option is turned off (`-DENABLE_PAW=0`).

However, when I set `-DENABLE_PAW=1`, I still encounter the following error:

> -- The CXX compiler identification is IntelLLVM 2024.2.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/intel/oneapi/compiler/2024.2/bin/icpx - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: /usr/bin/git (found version "2.34.1") 
-- Found git: attempting to get commit info...
-- Current commit hash: 00ae81d65
-- Last commit date: Thu Jul 11 13:31:52 2024 +0800
-- Found Cereal: /usr/include  
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.2") 
-- Found ELPA: /usr/local/lib/libelpa_openmp.so  
-- Performing Test ELPA_VERSION_SATISFIES
-- Performing Test ELPA_VERSION_SATISFIES - Success
-- Found MPI_CXX: /opt/intel/oneapi/mpi/2021.13/lib/libmpicxx.so (found version "3.1") 
-- Found MPI: TRUE (found version "3.1")  
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found OpenMP_CXX: -fiopenmp  
-- Found OpenMP: TRUE   
-- Looking for a CUDA compiler
-- Looking for a CUDA compiler - /usr/local/cuda-12.1/bin/nvcc
-- CUDA components detected, but USE_CUDA is set to OFF. NOT building CUDA version of ABACUS.
-- Upgrade to CMake version 3.25.2 or later for native support of Intel compiler(s) icpx. You are running version 3.22.1.
-- MKL_VERSION: 2024.2.0
-- MKL_ROOT: /opt/intel/oneapi/mkl/2024.2
-- MKL_SYCL_ARCH: None, set to ` intel64` by default
-- MKL_ARCH: None, set to ` intel64` by default
-- MKL_SYCL_LINK: None, set to ` dynamic` by default
-- MKL_LINK: None, set to ` dynamic` by default
-- MKL_SYCL_INTERFACE_FULL: intel_lp64
-- MKL_INTERFACE_FULL: intel_lp64
-- MKL_SYCL_THREADING: None, set to ` tbb_thread` by default
-- MKL_THREADING: None, set to ` intel_thread` by default
-- MKL_MPI: None, set to ` intelmpi` by default
-- Experimental oneMKL Data Fitting SYCL API does not support LP64 on CPU
-- Found /opt/intel/oneapi/mkl/2024.2/lib/libmkl_scalapack_lp64.so
-- Found /opt/intel/oneapi/mkl/2024.2/lib/libmkl_cdft_core.so
-- Found /opt/intel/oneapi/mkl/2024.2/lib/libmkl_intel_lp64.so
-- Found /opt/intel/oneapi/mkl/2024.2/lib/libmkl_intel_thread.so
-- Found /opt/intel/oneapi/mkl/2024.2/lib/libmkl_core.so
-- Found /opt/intel/oneapi/mkl/2024.2/lib/libmkl_blacs_intelmpi_lp64.so
-- Found /opt/intel/oneapi/mkl/2024.2/lib/libmkl_sycl_blas.so
-- Found /opt/intel/oneapi/mkl/2024.2/lib/libmkl_sycl_lapack.so
-- Found /opt/intel/oneapi/mkl/2024.2/lib/libmkl_sycl_dft.so
-- Found /opt/intel/oneapi/mkl/2024.2/lib/libmkl_sycl_sparse.so
-- Found /opt/intel/oneapi/mkl/2024.2/lib/libmkl_sycl_data_fitting.so
-- Found /opt/intel/oneapi/mkl/2024.2/lib/libmkl_sycl_rng.so
-- Found /opt/intel/oneapi/mkl/2024.2/lib/libmkl_sycl_stats.so
-- Found /opt/intel/oneapi/mkl/2024.2/lib/libmkl_sycl_vm.so
-- Found /opt/intel/oneapi/mkl/2024.2/lib/libmkl_tbb_thread.so
-- Found /opt/intel/oneapi/compiler/2024.2/lib/libiomp5.so
-- Found Torch: /home/itztony/Softwares/libtorch/lib/libtorch.so  
-- Updating submodules
-- Updating submodules
-- Updating submodules
-- The C compiler identification is IntelLLVM 2024.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/intel/oneapi/compiler/2024.2/bin/icx - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- The Fortran compiler identification is Intel 2021.0.0.20240602
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Check for working Fortran compiler: /opt/intel/oneapi/compiler/2024.2/bin/ifort - skipped
-- Checking for one of the modules 'libxc'
-- Found Libxc: /usr/lib/x86_64-linux-gnu/libxc.so  
-- Found Libxc: version 5.1.7
-- Configuring done
**CMake Error at deps/libpaw_interface/test/CMakeLists.txt:74 (add_executable):
  Target "paw_lib" links to target "IntelMKL::MKL" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?**
-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.

**Comment by caic99** [comment]
> I have successfully upgraded my MKL version to 2024.2.0 and pulled the latest repository of ABACUS. I am now able to compile ABACUS successfully, but only when the LIB_PAW option is turned off (`-DENABLE_PAW=0`).

@AsTonyshment Glad to hear that! The problem you've mentioned is fixed in the latest commit.

**Comment by Cstandardlib** [comment]
I successfully built commit b85c47 on Intel OneAPI 2024.1 & 2024.2 with -DBUILD_TESTING=ON -DENABLE_LIBRI=ON.
Build with deepks failed with errors:
![error](https://github.com/deepmodeling/abacus-develop/assets/49788094/e67aa4d8-c18f-427e-9af8-aba788ca930c)

**Comment by Qianruipku** [comment]
> @Qianruipku Would you try add a line `set(ENABLE_SCALAPACK ON)` in `CMakeLists.txt` before the line `set(MKL_INTERFACE lp64)`? Many Thanks.

still not work.

**Comment by caic99** [comment]
> Build with deepks failed with errors:

@Cstandardlib That seems to be another issue about building unit tests.

**Comment by caic99** [comment]
> @Qianruipku @AsTonyshment Hi there, would you try if #4644 patching works? Thanks.

@Qianruipku How about the modifications in #4644?

**Comment by Qianruipku** [comment]
> @Qianruipku @AsTonyshment Hi there, would you try if #4644 patching works? Thanks.

@caic99  do not find pdtran_
![image](https://github.com/deepmodeling/abacus-develop/assets/76200646/0d8f0330-76c4-46c1-803a-eb492f394990)

**Comment by caic99** [comment]
> do not find pdtran_

Use `icpx` in replace of `icpc`?

**Comment by Qianruipku** [comment]
> > do not find pdtran_
> 
> Use `icpx` in replace of `icpc`?

still
![image](https://github.com/deepmodeling/abacus-develop/assets/76200646/736c4e36-d53d-4105-b57b-37f7735184d4)

**Comment by caic99** [comment]
@Qianruipku 😢 The last thing to try is updating your oneAPI version, or using the latest developing image - I believe this solves your problem. Sorry for the inconvenience, and I've tried my best to make it compatible among environments.

**Comment by caic99** [comment]
@Qianruipku Hi Qianrui, 
I have some clues on the problem you've encountered. Would you mind we discuss them in #4657? Thanks.
